src/conduit/helpers/form.cljs

```diff
@@ -0,0 +1,25 @@
+(ns conduit.helpers.form
+  (:import goog.format.EmailAddress))
+
+(defn not-empty? [v]
+  (cond
+    (nil? v) false
+    (= "" v) false
+    :else true))
+
+(defn email? [v]
+  (.isValid (new EmailAddress v)))
+
+(defn validate! [form-data form-errors validators]
```

Comment:
This function is still overly complex. It shouldn't know about any state (pass in data instead of atoms). Instead of doing side effects just transform and return data. There should not be any side effectful functions like `doseq`, `doall`, etc.